service/frontend.js

```diff
@@ -102,10 +102,13 @@ module.exports = class WebFrontend extends events.EventEmitter {
 
         this._app.use((req, res, next) => {
             let port = ':' + this._app.get('port');
-            if (Config.HAS_REVERSE_PROXY ||
-                (req.protocol === 'http' && port === ':80') ||
-                (req.protocol === 'https' && port === ':443'))
+            if (Config.HAS_REVERSE_PROXY) {
+                let tmp = req.headers['x-forwarded-host'].split(':');
+                port = tmp[1] ? tmp[1] : '';
```

Comment:
```suggestion
                port = tmp[1] ? (':' + tmp[1]) : '';
```

Please test this in a real build before merging.

Reply:
I added some more controls over the headers provided by the RP.